src/output/gtk-shell.cpp

```diff
@@ -147,7 +147,18 @@ static void handle_gtk_shell_set_startup_id(wl_client *client, wl_resource *reso
  *  A view could use this to invoke the system bell, be it aural, visual or none at all.
  *  Not implemented.
  */
-static void handle_gtk_shell_system_bell(wl_client *client, wl_resource *resource, wl_resource *surface) {}
+static void handle_gtk_shell_system_bell(wl_client *client, wl_resource *resource, wl_resource *surface)
+{
+    auto wl_surface = static_cast<wl_resource*> (wl_resource_get_user_data(surface));
```

Comment:
This entire function needs a null pointer guard on the `surface` parameter, and should somehow handle a case for firing a bell signal to the shell itself, or something else. I fixed a crash simply adding a null guard to my local copy, but that doesn't send a bell signal anywhere in that case.

Reply:
seeing this, maybe it makes more sense to just emit the signal on core?